source/extensions/stat_sinks/hystrix/hystrix.h

```diff
@@ -0,0 +1,144 @@
+
+  /**
+   * Increment pointer of next value to add to rolling window.
+   */
+  void incCounter() { current_index_ = (current_index_ + 1) % window_size_; }
+
+  /**
+   * Generate the streams to be sent to hystrix dashboard.
+   */
+  void getClusterStats(std::stringstream& ss, absl::string_view cluster_name,
+                       uint64_t max_concurrent_requests, uint64_t reporting_hosts,
+                       uint64_t rolling_window);
+
+  /**
+   * Calculate values needed to create the stream and write into the map.
+   */
+  void updateRollingWindowMap(Upstream::ClusterInfoConstSharedPtr cluster_info,
+                              Stats::Store& stats);
+  /**
+   * Clear map.
+   */
+  void resetRollingWindow();
+
+  /**
+   * Return string represnting current state of the map. for DEBUG.
+   */
+  const std::string printRollingWindow() const;
+
+  /**
+   * Get the statistic's value change over the rolling window time frame.
+   */
+  uint64_t getRollingValue(absl::string_view cluster_name, absl::string_view stat);
+
+private:
+  /**
+   * Format the given key and absl::string_view value to "key"="value", and adding to the
+   * stringstream.
+   */
+  void addStringToStream(absl::string_view key, absl::string_view value, std::stringstream& info);
+
+  /**
+   * Format the given key and uint64_t value to "key"=<string of uint64_t>, and adding to the
+   * stringstream.
+   */
+  void addIntToStream(absl::string_view key, uint64_t value, std::stringstream& info);
+
+  /**
+   * Format the given key and value to "key"=value, and adding to the stringstream.
+   */
+  void addInfoToStream(absl::string_view key, absl::string_view value, std::stringstream& info);
+
+  /**
+   * Generate HystrixCommand event stream.
+   */
+  void addHystrixCommand(std::stringstream& ss, absl::string_view cluster_name,
+                         uint64_t max_concurrent_requests, uint64_t reporting_hosts,
+                         uint64_t rolling_window);
+
+  /**
+   * Generate HystrixThreadPool event stream.
+   */
+  void addHystrixThreadPool(std::stringstream& ss, absl::string_view cluster_name,
+                            uint64_t queue_size, uint64_t reporting_hosts, uint64_t rolling_window);
+
+  /**
+   * Building lookup name map for all specific cluster values.
+   */
+  void CreateCounterNameLookupForCluster(const std::string& cluster_name);
+
+  RollingStatsMap rolling_stats_map_;
+  uint64_t current_index_;
+  const uint64_t window_size_;
+  // TODO(trabetti): do we want this to be configurable through the HystrixSink in config file?
+  static const uint64_t DEFAULT_NUM_OF_BUCKETS = 10;
+  std::map<std::string, std::map<std::string, std::string>> counter_name_lookup;
+};
+
+typedef std::unique_ptr<HystrixStatCache> HystrixStatCachePtr;
+
+namespace Hystrix {
+
+class HystrixSink : public Stats::Sink, public Logger::Loggable<Logger::Id::hystrix> {
+public:
+  HystrixSink(Server::Instance& server, uint64_t num_of_buckets);
```

Comment:
output params 'server' should be last: https://google.github.io/styleguide/cppguide.html#Output_Parameters